low-level/framework/messaging/egress_accumulated_msgs_processor.py

```diff
@@ -122,7 +128,12 @@ def run(self):
                             continue
                     if "sensor_response_type" in dict_msg["message"]:
                         logger.info(f"Publishing Accumulated Alert: {message}")
-                    self._producer.send([message])
+                    if isinstance(self._producer, MessageProducer):
+                        self._producer.send([message])
+                        logger.error(f"Published Accumulated Message {message}")
+                        self.store_queue.delete()
+                    else:
+                        self.create_MsgProducer_obj()
```

Comment:
if code reaches here, then current message wont be procedded rt !? until all existing queue messages are iterated through

Reply:
yes, right.

Reply:
and picked up again in next iteration. so order of message delivery from accumulated queue is disturbed. we need to maintain the order

Reply:
message delivery from accumulated queue will not disturbed,  because we delete the msg only in case of successful send.
eg: if there are 3 msgs in queue m1,m2,m3,
 first we read msg from queue - m1, if it is not send successfully in 2nd iteration also we will read m1 only.
 and whatever new alert msgs are detected by sspl that will get added into accumulated queue because we  in egress_processor have check that if queue is not empty msg will be added into queue, 
 this way we are maintaining order.
 
 @sandeepsgt please correct me if I am wrong.

Reply:
Correct

Reply:
understood. as discussed, am assuming this is tested by now